src/biome.rs

```diff
@@ -1,31 +1,53 @@
-use std::{env, fs};
-use zed_extension_api::{self as zed, LanguageServerId, Result};
+use std::{env, fs, path::Path};
+use zed_extension_api::{self as zed, serde_json, LanguageServerId, Result};
 
-const SERVER_PATH: &str = "node_modules/.bin/biome";
+const SERVER_PATH: &str = "node_modules/@biomejs/biome/bin/biome";
 const PACKAGE_NAME: &str = "@biomejs/biome";
 
-struct BiomeExtension {
-  did_find_server: bool,
-}
+struct BiomeExtension;
 
 impl BiomeExtension {
-  fn server_exists(&self) -> bool {
-    fs::metadata(SERVER_PATH).map_or(false, |stat| stat.is_file())
+  fn server_exists(&self, path: &str) -> bool {
+    fs::metadata(path).map_or(false, |stat| stat.is_file())
   }
 
-  fn server_script_path(&mut self, language_server_id: &LanguageServerId) -> Result<String> {
-    let server_exists = self.server_exists();
-    if self.did_find_server && server_exists {
-      return Ok(SERVER_PATH.to_string());
+  fn server_script_path(
+    &mut self,
+    language_server_id: &LanguageServerId,
+    worktree: &zed::Worktree,
+  ) -> Result<String> {
+    // This is a workaround, as reading the file from wasm doesn't work.
+    // Instead we try to read the `package.json`, see if `@biomejs/biome` is installed
+    let package_json = worktree
+      .read_text_file("package.json")
+      .unwrap_or(String::from(r#"{}"#));
+    let package_json: Option<serde_json::Value> = serde_json::from_str(package_json.as_str()).ok();
+
+    let server_package_exists = package_json.is_some_and(|f| {
+      !f["dependencies"]["@biomejs/biome"].is_null()
+        || !f["devDependencies"]["@biomejs/biome"].is_null()
```

Comment:
Wouldn't this error if `dependencies` or `devDependencies` fields **aren't** aren't inside the `package.json`?

Reply:
I tested a bit, it will return null if one of the fields is non-existent. But feel free to test it again.